docs/spec_draft.md

```diff
@@ -753,6 +754,183 @@ defined and one of the following:
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.gather
+
+### Semantics
```

Comment:
My usual routine when I need to understand gather in-depth is: 1) realize that my memory has been wiped clean since the last time, 2) open https://www.tensorflow.org/xla/operation_semantics#gather, 3) struggle with correlating prose and pictures, while making notes on paper, 4) figure things out sometime later.

This PR is an improvement for me, because I liked the diagram a lot! With the diagram, the intermediate steps in index computations are explicitly written down, so I didn't even need notes to understand everything.

I didn't find prose in the Semantics section very helpful though, because the diagram was already mostly good enough. Therefore, I refactored the prose, made some changes to the diagram and [sent a proposal PR to your branch](https://github.com/sdasgup3/stablehlo/pull/3).

Here's how the spec looks like with the proposed update: https://github.com/burmako/stablehlo/blob/gather/docs/spec_draft.md#stablehlogather.

Reply:
Thanks for the update.